### DISCUSSION TITLE
How to clarify VAT exemption in ZUGFeRD 2.2

### DISCUSSION THREAD

**Discussion by CuNBecker** [stephanstapel/ZUGFeRD-csharp#444] [Q&A]
Hi,
I'm trying to implement ZUGFeRD invoices into our software. The software customer's services are exempt from VAT due to a specific case defined in german VAT law.  How do i specify this in ZUGFeRD 2.2? I've set TaxType to VAT and TaxCategoryCode to E but the TaxExemptionReasonCode-Enum does not offer me anything that fits a "regulated by german/national law" type of reason.  And none of the listed EU-rules really fit. 
How do i go about this?

**Comment by stephanstapel** [comment]
What special case is this about?

**Reply by CuNBecker** [reply]
§4 Nr.23 UStG

**Comment by fisfro** [comment]
Hi,
where you able, to define any exemption code, and will this code and the reason listet be part of the xml-file? 
I have done it like this:
desc.AddApplicableTradeTax(taxInfo.SteuerBasisBetrag, taxInfo.Steuersatz, TaxTypes.VAT, taxInfo.TaxCode,null,TaxExemptionReasonCodes.VATEX_EU_I,"local law"); 

but no exemption info will be included in the xml-file, althoug the tax object will includes the code and reason text.

Further I have detected that the codes in the class of the latest version will not filt to the vatex code list as seen on https://docs.peppol.eu/poacc/billing/3.0/codelist/vatex/. 

If I will add  VATEX_EU_I code as listed in the class in the xml file - I will get an validation error. If I change to VATEX-EU-I according to the current code list it will work:
      <cac:TaxCategory>
        <cbc:ID>E</cbc:ID>
        <cbc:Percent>0.00</cbc:Percent>
    	<cbc:TaxExemptionReasonCode>VATEX-EU-I</cbc:TaxExemptionReasonCode>
    	<cbc:TaxExemptionReason>local law</cbc:TaxExemptionReason>
        <cac:TaxScheme>
          <cbc:ID>VAT</cbc:ID>
        </cac:TaxScheme>
      </cac:TaxCategory>

So, if I would be able, that the exemption code will be part of the xml file (which is not the case now) , the xml will not pass the validation with the current VATEX Codes- I think.

CIAO Robert
